src/evm-block-extractor/tests/big_query_client_it.rs

```diff
@@ -0,0 +1,325 @@
+            let mut insert_request = TableDataInsertAllRequest::new();
+            insert_request.add_row(None, Row { name }).unwrap();
+
+            self.client
+                .tabledata()
+                .insert_all(
+                    &self.project_id,
+                    &self.dataset_id,
+                    &self.table_id,
+                    insert_request,
+                )
+                .await
+                .unwrap();
+        }
+
+        pub async fn get_rows(&self) -> Vec<String> {
+            let mut rs = self
+                .client
+                .job()
+                .query(
+                    &self.project_id,
+                    QueryRequest::new(format!(
+                        "SELECT * FROM `{}.{}.{}`",
+                        &self.project_id, &self.dataset_id, &self.table_id
+                    )),
+                )
+                .await
+                .unwrap();
+
+            let mut rows: Vec<String> = vec![];
+            while rs.next_row() {
+                let name = rs.get_string_by_name(NAME_COLUMN).unwrap().unwrap();
+                rows.push(name)
+            }
+            rows
+        }
+    }
+
+    async fn create_table(client: &Client, dataset: &Dataset) {
+        dataset
+            .create_table(
+                client,
+                Table::from_dataset(
+                    dataset,
+                    TABLE_ID,
+                    TableSchema::new(vec![TableFieldSchema::string(NAME_COLUMN)]),
+                ),
+            )
+            .await
+            .unwrap();
+    }
+}
+
+#[tokio::test]
+async fn test_big_query_stub() {
+    let docker = Cli::default();
+    let project_id = format!("test_project_{}", rand::random::<u64>());
+    let (gcp_client, _node, _temp_file, _auth) =
+        client::new_bigquery_client(&docker, &project_id).await;
+
+    let bq = BQ::new(gcp_client, &project_id).await;
+    let name = "foo";
+    bq.insert_row(name.to_string()).await;
+    let rows = bq.get_rows().await;
+    assert_eq!(rows, vec![name]);
+    println!("That's all Folks!");
+    bq.delete_dataset().await;
+}
+
+#[tokio::test]
+async fn test_insertion_of_blocks_and_retrieval_in_bq() {
+    let docker = Cli::default();
+    let project_id = format!("test_project_{}", rand::random::<u64>());
+    let (gcp_client, _node, _temp_file, _auth) =
+        client::new_bigquery_client(&docker, &project_id).await;
+    let dataset_id = format!("test_{}", rand::random::<u64>());
+
+    // Create dataset
+    gcp_client
+        .dataset()
+        .create(Dataset::new(&project_id, &dataset_id))
+        .await
+        .unwrap();
+
+    // bq::create_block_table(&gcp_client, &dataset).await;
+
+    let mut blockchain = Box::new(
+        BigQueryBlockChain::new_with_client(
+            project_id.clone(),
+            dataset_id.clone(),
+            gcp_client.clone(),
+        )
+        .unwrap(),
+    );
+
+    blockchain.init().await.unwrap();
```

Comment:
Add a test where the init is executed in a row to be sure it does not fail if the tables exist